internal/services/formatters/generic/semgrep/formatter_test.go

```diff
@@ -21,134 +21,131 @@ import (
 
 	"github.com/stretchr/testify/assert"
 
-	cliConfig "github.com/ZupIT/horusec/config"
-	"github.com/ZupIT/horusec/internal/entities/workdir"
+	"github.com/ZupIT/horusec/config"
 	"github.com/ZupIT/horusec/internal/services/formatters"
 )
 
 func TestParseOutput(t *testing.T) {
 	t.Run("Should return 1 vulnerabilities with no errors", func(t *testing.T) {
 		dockerAPIControllerMock := testutil.NewDockerMock()
 		dockerAPIControllerMock.On("SetAnalysisID")
-		analysis := &entitiesAnalysis.Analysis{}
-		config := &cliConfig.Config{}
-		config.WorkDir = &workdir.WorkDir{}
+		dockerAPIControllerMock.On("CreateLanguageAnalysisContainer").Return(output, nil)
 
-		output := "{ \"results\":[ { \"check_id\":\"python.lang.correctness.useless-comparison.no-strings-as-booleans\"," +
-			" \"path\":\"bad/vulpy.py\", \"start\":{ \"line\":36, \"col\":1 }, \"end\":{ \"line\":37, \"col\":23 }, " +
-			"\"extra\":{ \"message\":\"Using strings as booleans in Python has unexpected results.\\n`\\\"one\\\" and " +
-			"\\\"two\\\"` will return \\\"two\\\".\\n`\\\"one\\\" or \\\"two\\\"` will return \\\"one\\\".\\n In Python" +
-			", strings are truthy, evaluating to True.\\n\", \"metavars\":{ }, \"metadata\":{ }, \"severity\":\"ERROR\"" +
-			", \"lines\":\"if csp:\\n    print('CSP:', csp)\" } } ] }"
+		analysis := new(analysis.Analysis)
 
-		dockerAPIControllerMock.On("CreateLanguageAnalysisContainer").Return(output, nil)
+		cfg := config.New()
 
-		service := formatters.NewFormatterService(analysis, dockerAPIControllerMock, config)
+		service := formatters.NewFormatterService(analysis, dockerAPIControllerMock, cfg)
 		formatter := NewFormatter(service)
-
 		formatter.StartAnalysis("")
+
 		assert.Len(t, analysis.AnalysisVulnerabilities, 1)
 	})
 
 	t.Run("Should return 1 vulnerabilities with no errors", func(t *testing.T) {
 		dockerAPIControllerMock := testutil.NewDockerMock()
 		dockerAPIControllerMock.On("SetAnalysisID")
-		analysis := &entitiesAnalysis.Analysis{}
-		config := &cliConfig.Config{}
-		config.WorkDir = &workdir.WorkDir{}
-
-		output := "{ \"results\":[ { \"check_id\":\"python.lang.correctness.useless-comparison.no-strings-as-booleans\"," +
-			" \"path\":\"bad/vulpy.py\", \"start\":{ \"line\":36, \"col\":1 }, \"end\":{ \"line\":37, \"col\":23 }, " +
-			"\"extra\":{ \"message\":\"Using strings as booleans in Python has unexpected results.\\n`\\\"one\\\" and " +
-			"\\\"two\\\"` will return \\\"two\\\".\\n`\\\"one\\\" or \\\"two\\\"` will return \\\"one\\\".\\n In Python" +
-			", strings are truthy, evaluating to True.\\n\", \"metavars\":{ }, \"metadata\":{ }, \"severity\":\"WARNING\"" +
-			", \"lines\":\"if csp:\\n    print('CSP:', csp)\" } } ] }"
-
 		dockerAPIControllerMock.On("CreateLanguageAnalysisContainer").Return(output, nil)
 
-		service := formatters.NewFormatterService(analysis, dockerAPIControllerMock, config)
-		formatter := NewFormatter(service)
+		analysis := new(analysis.Analysis)
```

Comment:
Variable collides with imported package name.
Variables and packages with the same name may cause confusion

Reply:
I don't see this as a problem here, we are just creating an analysis pointer. If we tried to access another object from analysis package we will get a compiler error so we are not introduce any bug here.

Reply:
I think that we should avoid this, shadow imports are a bad practice. The project linter would report this, but we skip test files.

![ex-func](https://user-images.githubusercontent.com/63246935/143230347-e6220e5f-0744-442d-9a4e-ff7c063124dc.png)
![go-critic](https://user-images.githubusercontent.com/63246935/143230352-f9c99214-6b56-47bf-ba2b-d336297306ce.png)

Reply:
Yeah, but what should we do here? Use `entity` instead `analysis` which doesn't have much meaning or use an alias on import `analysisentity` which will make the package name too big and confusing because other usages of this package just call `entity`. 

I don't see shadow as bad practices, in my opinion this linter should only warn on shadown of variables for the same type, which can definitely cause bugs. Shadown of variables of different types will cause a compile error if we try to use the wrong variable. 

What would be the best name for this variable?

Reply:
I Maybe we can change the package name in the Devkit to prevent this to happening. I will resolve this conversation and let this discussion for later.